cmd/treedb-document-service/main.go

```
@@ -38,6 +44,24 @@ func main() {
+	}
+	if *mutexProfileFraction > 0 {
+		runtime.SetMutexProfileFraction(*mutexProfileFraction)
+	}
+	if handler := optionalPprofHandler(*pprofAddr); handler != nil {
+		listener, err := net.Listen("tcp", *pprofAddr)
+		if err != nil {
+			log.Fatalf("Failed to listen for pprof on %s: %v", *pprofAddr, err)
+		}
+		go func() {
+			log.Printf("TreeDB Document Service pprof listening on http://%s/debug/pprof/", *pprofAddr)
+			if err := http.Serve(listener, handler); err != nil && !errors.Is(err, http.ErrServerClosed) {
```

golangci-lint (2.12.2)

[medium] G114: Use of net/http serve function that has no support for setting timeouts

(gosec)